### PR TITLE
remove pulp from the documentation

### PR DESCRIPTION
**Pull request by mlangsdorf**
part of osbs-5098

Remove all references to pulp and pulp related variables, except
for the pulp references in the historical overview of arrangements.

Signed-off-by: Mark Langsdorf <mlangsdo@redhat.com>

### CODE REVIEW

File: docs/users.rst [hunk 1]
```diff
@@ -428,8 +412,6 @@ modular_koji_tags
 must have a non-empty list of modules. The "packages" key, and only the "packages"
 key, can have an empty list.**
 
-**The "packages", "modules" and "pulp_repos" keys can be used mutually.**
```
Comment (athos-ribeiro) [inline docs/users.rst]: Should we keep the information that the package and the modules keys can be used mutually?

File: docs/users.rst [hunk 2]
```diff
@@ -302,8 +302,6 @@ An example::
     - nss_wrapper  # package name, not an NVR.
     - httpd
     - httpd-devel
-    # used for requesting ODCS compose of type "pulp"
-    pulp_repos: true
```
Comment (athos-ribeiro) [inline docs/users.rst]: Do we want to remove the pulp references from this file (`docs/users.rst`)? These options are the ones passed to ODCS for the composes.

File: docs/build_parameters.rst [hunk 3]
```diff
@@ -244,7 +244,7 @@ example, if **reactor_config** defines::
     - kojisecret
     - pulpsecret
 
-Secrets named **kojisecret** and **pulpsecret** must be available in orchestrator and
+Secrets named **kojisecret** must be available in orchestrator and
```
Comment (twaugh) [inline docs/build_parameters.rst]: Now there's just one of them there, so this should probably read `A secret named ...`, and further down, `It doesn't need ...`.

File: docs/users.rst (hunk 2 above)
Comment (twaugh) [reply docs/users.rst]: We need to keep the `pulp_repos` option. Please revert this part.

File: docs/users.rst [hunk 4]
```diff
@@ -382,12 +380,6 @@ packages
   in ``container.yaml``, you can use the empty list to just make everything
   available.
 
-pulp_repos
-  boolean to control whether or not an ODCS compose of type "pulp" should be
-  requested. If set to true, ``content_sets.yml`` must also be provided. A
-  compose will be requested for each architecture in ``content_sets.yml``.
-  See :ref:`content_sets.yml`.
-
```
Comment (twaugh) [inline docs/users.rst]: We need to keep this part.

File: docs/users.rst [hunk 5]
```diff
@@ -402,14 +394,6 @@ inherit
...
   inheritance and false is always assumed.
 
-include_unpublished_pulp_repos
-  If you set ``include_unpublished_pulp_repos: true`` under the ``compose``
-  section in ``container.yaml``, the ODCS composes can pull from unpublished
-  pulp repositories. The default is ``false``. Use this setting to make
-  pre-release RPMs available to your container images. Use caution with this
-  setting, because you could end up publicly shipping container images with
-  RPMs that you have not exposed publicly otherwise.
-
```
Comment (twaugh) [inline docs/users.rst]: We need to keep this part.

File: docs/users.rst (hunk 1 above)
Comment (twaugh) [reply docs/users.rst]: We need to keep this part altogether.

The `pulp_repos` option is about RPMs stored in Pulp, not container images.

File: docs/users.rst [hunk 6]
```diff
@@ -586,9 +568,7 @@ Content Sets
 ------------
 
 The file ``content_sets.yml`` is used to define the content sets relevant to the
-container image.  This is relevant if RPM packages in container image are in
-pulp repositories. See ``pulp_repos`` in :ref:`container.yaml-compose` for how
-this file is used during build time.
+container image.
```
Comment (twaugh) [inline docs/users.rst]: We need to keep this part.

File: docs/build_parameters.rst (hunk 3 above)
Comment (mlangsdorf) [reply docs/build_parameters.rst]: Fixed.

**Comment by mlangsdorf** [comment]
Reverted the changes involving pulp for ODCS composes.

**Pull request merged**